src/System.Net.WebSockets.Client/tests/SendReceiveTest.cs

```diff
@@ -86,55 +85,44 @@ public class SendReceiveTest : ClientWebSocketTestBase
             {
-                var cts = new CancellationTokenSource(TimeOutMilliseconds);
+                var tasks = new Task[2];
 
-                Task[] tasks = new Task[10];
+                // Need large buffer, or Send completes too quickly and no exception is thrown.
+                var sendBuffer = new byte[128 * 1024];
+                var segment = new ArraySegment<byte>(sendBuffer);
 
-                try
+                for (int i = 0; i < tasks.Length; i++)
                 {
-                    for (int i = 0; i < tasks.Length; i++)
-                    {
-                        tasks[i] = cws.SendAsync(
-                            WebSocketData.GetBufferFromText("hello"),
-                            WebSocketMessageType.Text,
-                            true,
-                            cts.Token);
-                    }
+                    tasks[i] = cws.SendAsync(
+                        segment,
+                        WebSocketMessageType.Text,
+                        true,
+                        CancellationToken.None);
+                }
 
-                    Task.WaitAll(tasks);
+                await Assert.ThrowsAsync<InvalidOperationException>(() => Task.WhenAll(tasks));
+                Assert.Equal(WebSocketState.Aborted, cws.State);
 
-                    Assert.Equal(WebSocketState.Open, cws.State);
-                }
-                catch (AggregateException ag)
+                Assert.All(tasks, task =>
                 {
-                    foreach (var ex in ag.InnerExceptions)
-                    {
-                        if (ex is InvalidOperationException)
-                        {
-                            Assert.Equal(
-                                ResourceHelper.GetExceptionMessage(
-                                    "net_Websockets_AlreadyOneOutstandingOperation",
-                                    "SendAsync"),
-                                ex.Message);
-
-                            Assert.Equal(WebSocketState.Aborted, cws.State);
-                        }
-                        else if (ex is WebSocketException)
-                        {
-                            // Multiple cases.
-                            Assert.Equal(WebSocketState.Aborted, cws.State);
-
-                            WebSocketError errCode = (ex as WebSocketException).WebSocketErrorCode;
-                            Assert.True(
-                                (errCode == WebSocketError.InvalidState) || (errCode == WebSocketError.Success),
-                                "WebSocketErrorCode");
-                        }
-                        else
-                        {
-                            Assert.IsAssignableFrom<OperationCanceledException>(ex);
-                        }
-                    }
-                }
+                    Assert.Equal(TaskStatus.Faulted, task.Status);
```

Comment:
This does not look like a safe change.  Due to timing and race conditions and the like, it's possible that any number of the operations could actually complete successfully.  Seems like this is going to be very prone to sporadic failures in the lab.